.github/workflows/python_macos_test.yml

```diff
@@ -0,0 +1,165 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: Build and test using Maven
+
+on:
+  workflow_call:
+    inputs:
+      java:
```

Comment:
it is a bit more complicated than I thought, it seems we simply specify the java/hadoop version / branches in other python daily jobs.

https://github.com/apache/spark/blob/9d79ab42b127d1a12164cec260bfbd69f6da8b74/.github/workflows/build_python_3.10.yml#L34

Reply:
or should we specify the versions in another file?

Reply:
1. Only the content defined here can specify versions in other scripts. We can see that `build_and_test.yml` also defines this content.
2. Actually, for content that is the same as the default version, it is not necessary to specify it again in the sub-script."So it's okay not to specify 

```
     java: 17
      branch: master
      hadoop: hadoop3
``` 

again in `build_python_3.10.yml`